### PR TITLE
Introduce Kustomize and variants (scos/fcos)

### PR DESCRIPTION
**Pull request by aleskandro**
This PR adds kustomize to ease the management of both the buildconfigs and the deployment of the Tekton objects.

In particular:

- The kustomization for the buildconfigs replaces the builder images with the okd builder from  https://github.com/okd-project/images
- The kustomization adds the management of three requests and limits classes, a default, a high, and a low resources profiles. These have to be tuned yet.
- A new parameter, max_concurrent_builds, let the user able to set how many builds to run in parallel, based on the available resources. Together with the appropriate requests and limits profiles set in the buildconfigs, this change aims to optimize the batch-build task and to guarantee more concurrency safety when gaining resources to build the payload.
- Finally, the PR introduces two kustomize overlay variants for OKD/FCOS and OKD/SCOS.

### CODE REVIEW

File: buildconfigs/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+              name: "release:builder"
+           as:
+              - 'registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.18-openshift-4.12'
+              - 'registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.18-openshift-4.11'
+              - 'registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.17-openshift-4.11'
+              - 'registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.17-openshift-4.10'
+              - 'registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-4.10'
+              - 'registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-4.9'
+              - 'registry.ci.openshift.org/ocp/builder:rhel-8-etcd-golang-1.16'
+              - 'registry.ci.openshift.org/openshift/release:golang-1.17'
```
Comment (vrutkovs) [inline buildconfigs/kustomization.yaml:22]: Not quite sure what this does - adds these images as base to all buildconfigs?

File: buildconfigs/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+  - patch: |-
+      - op: add
+        path: "/spec/resources"
+        value:
+          limits:
+            memory: 2Gi
+            cpu: 2000m
+          requests:
+            memory: 1Gi
+            cpu: 1000m
```
Comment (vrutkovs) [inline buildconfigs/kustomization.yaml:35]: This is brilliant, I think we want some buildconfigs labelled as "large" too (hyperkube rebuild may use even more)

File: forks/cluster-node-tuning-operator/Dockerfile.centos9 [hunk 3]
```diff
@@ -14,9 +14,7 @@ RUN INSTALL_PKGS=" \
     LC_COLLATE=C cat ../patches/*.diff | patch -Np1 && \
     dnf build-dep tuned.spec -y && \
     make rpm PYTHON=/usr/bin/python3 && \
-    rm -rf /root/rpmbuild/RPMS/noarch/{tuned-gtk*,tuned-utils*,tuned-profiles-compat*} && \
```
Comment (vrutkovs) [inline forks/cluster-node-tuning-operator/Dockerfile.centos9:17]: This change needs to be a separate PR

File: pipelines/build-from-scratch.yaml [hunk 4]
```diff
@@ -22,21 +21,25 @@ spec:
...
-        name: openshift-client
-        kind: ClusterTask
+        name: batch-build
       params:
-        - name: SCRIPT
-          value: |
-            oc start-build builder -w && \
-            oc start-build forked-dockerfiles -w && \
-            oc start-build base -w
+        - name: max_concurrent_builds
```
Comment (vrutkovs) [inline pipelines/build-from-scratch.yaml:26]: :+1:

File: buildconfigs/kustomization.yaml (hunk 1 above)
Comment (aleskandro) [reply buildconfigs/kustomization.yaml:22]: Hey @vrutkovs, 

this patch is to bulk replace the (not runtime-stage) FROM layers that link any of the images in the `as` array with the release:builder imagestreamtag.

Currently, we do that for each image in a fine-grained way, but maintaining each single buildconfig "as" stanza in sync with the upstream payload images is harder. I don't know if there is another way like the one we do for the runtime stage base image replacement.

So the patch changes any buildconfig not having the skip label to have this replacement rule. Only the buildconfigs that actually refer to images in the array will consume this. E.g., with the rebase landed yesterday, we will have to change some of the `as` stanza to replace `registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.1**9**-openshift-4.12` instead of `registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.18-openshift-4.12` (and we also miss golang 1.19 in cs9).

File: buildconfigs/kustomization.yaml (hunk 2 above)
Comment (aleskandro) [reply buildconfigs/kustomization.yaml:35]: Yes, I have some changes locally already to have a few classes of resource requirements to address what you say

File: buildconfigs/kustomization.yaml (hunk 1 above)
Comment (LorbusChris) [reply buildconfigs/kustomization.yaml:23]: It'd be good to add a comment to this file explaining this.

Comment (aleskandro) [reply buildconfigs/kustomization.yaml:26]: Done

**Review by LorbusChris — APPROVED** [review]
